client-spec/client-specification-v1.md

```diff
@@ -0,0 +1,316 @@
+
+This field can have the following values:
+- `stable`
+- `testing`
+- `development`
+
+This field may be use to restrict the list of installable packages
+(e.g, to only allow `stable` modules).
+
+If this field is missing, clients must assume its value to be `stable`.
+
+### ksp_version
+
+This field may contain either:
+- `any`
+- a string matching the regular expression:
+		[0-9]+\\.[0-9]+(\\.[0-9]+)?
+
+If the field contains the string `any`,
+clients must allow the installation of the module
+on any version of the game.
+
+Otherwise, the client must only allow the installation of the module
+into a copy of the game with a matching version number.
+
+### ksp_version_min and ksp_version_max
+
+Like the `ksp_version` field, these fields can contain either the string `any`
+or a version number.
+
+These fields are mutually exclusive with the `ksp_version` field.
+Clients should consider metadata files that contain both types of fields as invalid.
+In this case, it is suggested to enforce the most specific version constraint.
+
+
+## install stanzas
+
+Installation instructions are specified using the `install` field.
+This *optional* field contains an array of install directive
+(called "stanzas" in some of the documentation).
+Each stanza defines where to install a set of files.
+
+If no install sections are provided,
+a CKAN client must find the top-most directory in the archive that matches the module identifier,
+and install that with a target of `GameData`.
+
+### Required: file and install_to
+
+If an `install` stanza is present,
+it must contain at least two fields:
+
+- `file`: path of a file or folder, relative to the top-level of the zip file obtained through the `download` link. The file (or folder) must be extracted to the location specified by the `install_to` field.
+- `install_to`: the location where the file or folder must be extracted. This field can only take one of the following 4 values:
+	. `GameRoot`: the root folder of the KSP installation;
+	. `GameData`: equal to `$(GameRoot)/GameData`;
+    . `Ships`: equal to `$(GameRoot)/Ships`
+    . `Tutorial`: equal to #TODO: where!?
+
+The client must not allow the installation of any file in any directory other than those listed above.
+The file or folder located by the `file` field must be extracted to the destination directory
+preserving the directory tree structure in the zip file.
+Installation of any file outside of the game's directory is strictly forbidden.
+
+### Optional: filters
+
+If a `filter` field is specified,
+it contains one or more files that must be excluded from the installation
+even if they are selected by the corresponding `file` field.
+The `filter` field only affects files in the same install stanza
+(that is, install stanzas are independent from each other).
+
+Analogously, if a `filter_regexp` is specified
+the client must exclude from the installation
+all the files that match one or more of the listed regular expressions.
+
+### File overwriting

```

Comment:
We might want to remove this entry. Issues like https://github.com/KSP-CKAN/CKAN/issues/1311 show that we have thought about (and generally support) a limited method for overwriting so I don't see why we would impose this restriction on others.